### PR TITLE
Fixes the parser to work with test cases provided

### PR DESCRIPTION
**Pull request by pontuslaestadius**
https://github.com/Noahnom/json-test-DIT029

all test cases: needs modified diagramCheck to function.
test5: parallel diagram breaks.
test4: ok
test3: par breaks
test2: ok
test1: breaks the server due to malformed parsing because uppercases are not atom wrapped.

After fixing:
test1: ok
test2: ok
test3: ok
test4: ok
test5: ok

NOTE:
Test 3: Testing 2 stacked par frames. Second par frame is inside the first section of the first frame.
Not fixing this.

### CODE REVIEW

File: Server/src/diagramcoordinator.erl [hunk 1]
```diff
@@ -85,9 +86,16 @@ spawn_node(Class_name, Did, Coordinator) ->
...
 %sends a message to the given node
 send_message(Receiver, From, To, Message, To_pid, Message_number, Coordinator, Did) ->
-  Receiver ! {send_message, From, To, Message, To_pid, Message_number},
-  receive
-    {send_reply, From, To, Message, To_pid, Message_number} -> 
-		%Sends info to the Coordinator, that the node sucessfully sent a message. To be printed in the executionlog	
-		Coordinator ! {Did, print_information, ["Node " ++ atom_to_list(From) ++ " sent a message to " ++ atom_to_list(To)]} 
-		end.
+
+	% Checks if there is a valid pid provided.o
```
Comment (IsabelleTornqvist) [inline Server/src/diagramcoordinator.erl]: .o? Other than that, code looks good from what I can see. Haven't had the time to test it yet though, will do that as soon as possible

**Review by timjon — CHANGES_REQUESTED** [review]
The parsing does not work for test2.json, test3.json and test5.json

The parsing goes through but everything doesnt get parsed so some parts of the information disapears

**Review by ghost — APPROVED** [review]
Nice! you might as well move the diagram checker defect in the defect backlog to resolved :D

**Comment by pontuslaestadius** [comment]
Not going to be implemented.

**Pull request closed without merging**